### DISCUSSION TITLE
How to update the sensor version on an update policy

### DISCUSSION THREAD

**Discussion by jeff-a-holland** [CrowdStrike/falconpy#52] [Q&A] [answered]
I want to programmatically update the sensor version of an update policy that has auto updates turned off. The use cases is as follows:

- A test group with a handful of hosts is set to auto update, while the rest of the hosts in the org have auto updates off.
- When a new version is released, we monitor the test group and stage the org for an update a week later. We can track the versions using the class getSensorUpdatePoliciesV2. This works fine.
- When it's time to update the org hosts, I want to change the policy version via the API. Having problems with this part.

Q1: Do I use updateSensorUpdatePoliciesV2 or performSensorUpdatePoliciesAction?
Q2: And what is the format of the BODY string? 
Q3: If I am to use performSensorUpdatePoliciesAction, what is the "action" string?

I've tried to format the BODY string based off the output the JSON blob printed out from the getSensorUpdatePoliciesV2 response, but no luck. 

Thanks

**Comment by jshcodes** [comment, marked as answer]
These two operations perform different but similar functions.

1. __performSensorUpdatePolicyAction__ is taking action on the policy, such as enabling, or disabling it. This operation can be executed against a list of IDs, with the body payload also supporting additional parameters that can be passed to the requested action. __updateSensorUpdatePoliciesV2__ updates sensor update policies by specifying the ID of the policy and what details to update. This operation also provides additional support for uninstall protection. Based upon this detail, I believe you will want to look at _updateSensorUpdatePoliciesV2_.
2. Here are the generic body payload examples for these two operations:
    - updateSensorUpdatePoliciesV2
    ```json
	{
	"resources": [
		{
		"description": "string",
		"id": "string",
		"name": "string",
		"settings": {
			"build": "string",
			"uninstall_protection": "ENABLED"
		      }
		}
  	    ]
	}
    ```
    
    - performSensorUpdatePoliciesAction
    ```json
    {
      "action_parameters": [
        {
          "name": "string",
          "value": "string"
        }
      ],
      "ids": [
         "string"
       ]
     }
    ```
3. Available actions for performSensorUpdatePoliciesAction: 
    - add-host-group
    - disable
    - enable
    - remove-host-group 

> Note: While researching this answer for you, I believe I've discovered a programmatic issue with the performSensorUpdatePoliciesAction method preventing developers from being able to easily specify the action string on the URL.  I've submitted a bug report (#53) for this issue, we will have it resolved shortly. In the interim you may access this operation's functionality by making use of the Uber class.

**Reply by jeff-a-holland** [reply]
Wow, fast response. Thanks, just what I needed!!

**Reply by jshcodes** [reply]
Happy to help, let us know if you have any more questions! 😀
